patches/server/0891-Fire-EntityChangeBlockEvent-in-more-places.patch

```diff
@@ -110,6 +126,23 @@ index 68a8d3b16d49c10fc9834f32009095d35c9c55a8..1f8b7b50c6aa24778d87821ae2ff4d01
              if (player instanceof ServerPlayer) {
                  CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger((ServerPlayer)player, blockPos, itemStack);
              }
+diff --git a/src/main/java/net/minecraft/world/item/PotionItem.java b/src/main/java/net/minecraft/world/item/PotionItem.java
+index 0b344e20ad4093b135c58905fb82ffff05898bb2..904c0fc40152a0b8af5771a96e4bff9b44fb228c 100644
+--- a/src/main/java/net/minecraft/world/item/PotionItem.java
++++ b/src/main/java/net/minecraft/world/item/PotionItem.java
+@@ -107,6 +107,12 @@ public class PotionItem extends Item {
+         BlockState iblockdata = world.getBlockState(blockposition);
+ 
+         if (context.getClickedFace() != Direction.DOWN && iblockdata.is(BlockTags.CONVERTABLE_TO_MUD) && PotionUtils.getPotion(itemstack) == Potions.WATER) {
++            // Paper start
++            if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(entityhuman, blockposition, Blocks.MUD.defaultBlockState()).isCancelled()) {
++                ((net.minecraft.server.level.ServerPlayer) entityhuman).getBukkitEntity().updateInventory();
```

Comment:
Can this change to `entityhuman.containerMenu.sendAllDataToRemote()` just to not use the API method here? No reason to and sticking to native code is better here I think.

Reply:
yes sure but there some inconsistencies about that in the whole api